crates/oxc_linter/src/rules/eslint/class_methods_use_this.rs

```diff
@@ -0,0 +1,646 @@
+use std::borrow::Cow;
+
+use itertools::Itertools;
+use oxc_ast::{
+    AstKind,
+    ast::{AccessorProperty, Expression, PropertyDefinition, TSAccessibility},
+};
+use oxc_ast_visit::Visit;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNode;
+use oxc_span::{CompactStr, GetSpan, Span};
+
+use crate::{LintContext, rule::Rule};
+
+fn class_methods_use_this_diagnostic(span: Span, name: Option<Cow<'_, str>>) -> OxcDiagnostic {
+    let method_name_str = name.map_or(String::new(), |name| format!(" `{name}`"));
+    OxcDiagnostic::warn(format!("Expected method{method_name_str} to have this."))
+        .with_help(format!("Consider converting method{method_name_str} to a static method."))
+        .with_label(span)
+}
+
+#[derive(Debug, Clone)]
+pub struct ClassMethodsUseThis {
+    except_methods: Vec<MethodException>,
+    enforce_for_class_fields: bool,
+    ignore_override_methods: bool,
+    ignore_classes_with_implements: Option<IgnoreClassWithImplements>,
+}
```

Comment:
this should be boxed, to avoid the change in size of rule enum here

  assert_eq!(size_of::<RuleEnum>(), 32);